### DISCUSSION TITLE
Metrics related to kubernetes_logs rotate_wait_secs

### DISCUSSION THREAD

**Discussion by davidcpell** [vectordotdev/vector#22060] [Q&A] [answered]
I've recently turned on [rotate_wait_secs](https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#rotate_wait_secs) in my Kubernetes environments because I was running into issues where Vector would be holding tons of file handles for files that have passed rotation size but hasn't finished processing them due to the amount of work it's doing. This led to node DiskPressure over time and pod evictions.

Using this helped, but it'd be _really_ helpful to be able to see how often Vector is dropping file handles early and any possible metadata related to which files (container name, whatever). Is this available/something that could be reasonably added?

**Comment by jszwedko** [comment, marked as answer]
Mmm I don't think there are metrics around that at the moment. It would be useful though. Would you want to open a feature request?

**Reply by davidcpell** [reply]
Thanks for the reply. And yes I've just opened this Issue: https://github.com/vectordotdev/vector/issues/22143
